### PR TITLE
Flutter 2.2.0 Android

### PR DESCRIPTION
**Pull request by CTLalit**
Flutter Release: 2.2.0 (Android)
Android - 6.1.1

https://wizrocket.atlassian.net/browse/SDK-3723

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+-------------------------------------------
+**What's new**
+* **[Android Platform]**
+  * Supports [CleverTap Android SDK v6.1.1](https://github.com/CleverTap/clevertap-android-sdk/blob/master/docs/CTCORECHANGELOG.md#version-611-february-27-2024).
+
+* **[iOS Platform]**
+  * Supports [CleverTap iOS SDK v6.1.0](https://github.com/CleverTap/clevertap-ios-sdk/releases/tag/6.1.0).
+  * Adds privacy manifests.
+
+* **[Android and iOS Platform]**
```
Comment (Anush-Shand) [inline CHANGELOG.md:13]: These 2 points are not required in flutter docs as we never released a version supporting v6.1.0

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+  * Fixes an issue of incorrect endpoint in the case of network handshake. 
+  * Fixes a bug in Client Side InApps with regards to frequency limits.
+
+**Bug Fixes**
+* **[Android Platform]**
+  * Fixes InApps crash in a rare activity destroyed race condition 
+  * Fixes Potential ANR in a race condition of SDK initialisation in multithreaded setup 
+  * Fixes #456 - Build issues due to AGP 8
```
Comment (Anush-Shand) [inline CHANGELOG.md:21]: This point was native specific, can be removed

File: example/android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,7 +1,6 @@
 <manifest
     xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
-    package="com.example.clevertap_plugin_example">
```
Comment (Anush-Shand) [inline example/android/app/src/main/AndroidManifest.xml:4]: Removal of package needs to be tested for compatibility with lower versions of flutter. Will verify once on my end

File: example/android/build.gradle [hunk 4]
```diff
@@ -20,14 +8,20 @@ allprojects {
...
 
-rootProject.buildDir = '../build'
+rootProject.layout.buildDirectory = '../build'
+
 subprojects {
-    project.buildDir = "${rootProject.buildDir}/${project.name}"
+    project.layout.buildDirectory = "${rootProject.layout.buildDirectory}/${project.name}"
+    if (project.hasProperty("android")) {
+        if (project.android.hasProperty("namespace")) {
+            namespace 'com.clevertap.clevertap_plugin'
```
Comment (Anush-Shand) [inline example/android/build.gradle:17]: This namespace doesn't belong here? It is for the clevertap plugin and not the example app

File: example/lib/main.dart [hunk 5]
```diff
@@ -2359,6 +2358,7 @@ class _MyAppState extends State<MyApp> {
   }
 
   void activate() {
+    super.activate();
```
Comment (Anush-Shand) [inline example/lib/main.dart:2361]: Is this some bugfix?

**Comment by Anush-Shand** [comment]
Also need to test ClassNotFoundException https://wizrocket.atlassian.net/browse/SDK-3699 and whether it can be fixed on the plugin side. Else we need to update the docs to provide a workaround for the client

File: CHANGELOG.md (hunk 2 above)
Comment (CTLalit) [reply CHANGELOG.md:21]: done

File: example/android/build.gradle (hunk 4 above)
Comment (CTLalit) [reply example/android/build.gradle:17]: done.

File: example/lib/main.dart (hunk 5 above)
Comment (CTLalit) [reply example/lib/main.dart:2361]: it was a warning produced in build log

**Review by Anush-Shand — APPROVED** [review]
LGTM

**Pull request merged**